### PR TITLE
feat(preset-mini): support fontSize with letter-space

### LINKED ISSUE #2906: Support additional config `letter-spacing` inside `fontSize`
### Clear and concise description of the problem

Currently, UnoCSS supports specifying font size inside the fontSize theme config:
```js
theme: {
    fontSize: {
      sm: ['15px'],
    },
```
or additionally, also specify line-height:
```js
theme: {
    fontSize: {
      sm: ['15px', '1.5em'],
    },
```
Similar to Tailwind theme config, we should also allow an additional config for letter-spacing:
```js
theme: {
    fontSize: {
      sm: ['15px', '1.5em', '-0.004em'],
    },
```
This is helpful for typesetting for perfect looking text :)

### Suggested solution

Allow an optional additional config inside the array:

```js
theme: {
    fontSize: {
      sm: [size, line height, letter-spacing],
    },
```

```js
theme: {
    fontSize: {
      sm: ['15px', '1.5em', '-0.004em'],
    },
```

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Read the `README.md` of using the package.
- [X] Already used the [Interactive Docs](https://unocss.dev/interactive/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by zyyv**
close #2906, duplicate of #3087 

Some build conifg changes refer to #711

### CODE REVIEW

File: packages/core/src/utils/basic.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
+import type { Arrayify } from '../types'
+
 export function toArray<T>(value: T | T[] = []): T[] {
   return Array.isArray(value) ? value : [value]
 }
 
+export function toArrayReadonly<T>(value: T) {
+  return (Array.isArray(value) ? value : [value]) as Arrayify<T>
+}
```
Comment (antfu) [inline packages/core/src/utils/basic.ts:9]: I am not sure if to have a runtime utility that only does type assessment (increase bundle size but not helpful at runtime). Also I think the name `Arrayify` is a bit confusing as it does not indicate it's readonly part. Maybe a `Readonly<ToArray<T>>` would be more straightforward.

File: packages/preset-mini/build.config.ts [hunk 2]
```diff
@@ -13,5 +13,9 @@ export default defineBuildConfig({
   declaration: true,
   rollup: {
     emitCJS: true,
+    dts: {
+      respectExternal: false,
+    },
   },
+  failOnWarn: false,
```
Comment (antfu) [inline packages/preset-mini/build.config.ts:20]: Any reason these are added? `failOnWarn: false` feels dangerous

File: packages/core/src/utils/basic.ts (hunk 1 above)
Comment (antfu) [reply packages/core/src/utils/basic.ts]: Reverted them so I could merge this feature first. We could discuss about it somewhere else

File: packages/preset-mini/build.config.ts (hunk 2 above)
Comment (zyyv) [reply packages/preset-mini/build.config.ts]: I also feel is not good, but ci error. see https://github.com/unocss/unocss/actions/runs/6608185990/job/17946548533?pr=3149#step:9:277

File: packages/core/src/utils/basic.ts (hunk 1 above)
Comment (zyyv) [reply packages/core/src/utils/basic.ts]: tks ~

**Pull request merged**